### PR TITLE
#601 - Add underapproximate function

### LINKED ISSUE #601: Polyhedral underapproximation
Analogous to polyhedral overapproximations, we can easily underapproximate a **convex** set by sampling support vectors `v_1, ..., v_k` and taking the convex hull of those (in our terminology, `VPolytope([v_1, ..., v_k])`).
There might also be a procedure like our iterative refinement for epsilon-close underapproximation.

### PR DESCRIPTION
**Pull request by mforets**
Closes #601.

### CODE REVIEW

**Comment by mforets** [comment]
Yes. Actually the `VPolytope` constructor does not compute the convex hull ATM (this is https://github.com/JuliaReach/LazySets.jl/issues/1392). So my impementation is wrong and we should either return a `ConvexHullArray` of `Singleton` or just apply `convex_hull` to `vinner`.

**Comment by schillic** [comment]
> So my impementation is wrong

I do not see why. It may just contain redundant vertices.

File: src/Approximations/underapproximate.jl [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+                         the support vectors with a convex hull operation
+### Output
+
+The `VPolytope` obtained by taking the convex hull of the support vectors of `X`
+along the directions determined by `dirs`.
+"""
+function underapproximate(X::LazySet{N}, dirs::AbstractDirections;
+                          apply_convex_hull::Bool=false) where {N<:Real}
+    @assert LazySets.dim(X) == LazySets.dim(dirs)
+    vinner = Vector{Vector{N}}(undef, length(dirs)) # TODO could be generalized to eltype(dirs)
```
Comment (mforets) [inline src/Approximations/underapproximate.jl]: The reason that `eltype(dirs)` doesn't currently work here is that `eltype` of directions is other than `Vector` in general, while `VPolytope`'s `vertices` fields is a `vertices::Vector{Vector{N}}`. That should be eventually generalized.

File: src/Approximations/underapproximate.jl [hunk 2]
```diff
@@ -1,20 +1,29 @@
...
 The `VPolytope` obtained by taking the convex hull of the support vectors of `X`
 along the directions determined by `dirs`.
 """
-function underapproximate(X::LazySet{N}, dirs::AbstractDirections) where {N<:Real}
-    @assert dim(X) == dim(dirs)
-    vinner = [σ(di, X) for di in dirs]
+function underapproximate(X::LazySet{N}, dirs::AbstractDirections;
+                          apply_convex_hull::Bool=false) where {N<:Real}
+    @assert LazySets.dim(X) == LazySets.dim(dirs)
+    vinner = Vector{Vector{N}}(undef, length(dirs)) # TODO could be generalized to eltype(dirs)
```
Comment (schillic) [inline src/Approximations/underapproximate.jl]: Not sure about the TODO. The result type of `σ` also depends on `X`.

File: src/Approximations/underapproximate.jl (hunk 1 above)
Comment (schillic) [reply src/Approximations/underapproximate.jl]: Not sure about the TODO. The result type of σ also depends on X.

**Pull request merged**